packages/salesforcedx-vscode-core/src/commands/functions/forceFunctionStart/ForceFunctionContainerStartExecutor.ts

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Benny, getFunctionsBinary } from '@heroku/functions-core';
+import * as vscode from 'vscode';
+import { channelService } from '../../../channels';
+import { nls } from '../../../messages';
+import { notificationService } from '../../../notifications';
+import { telemetryService } from '../../../telemetry';
+import { FunctionService } from '../functionService';
+import { FUNCTION_RUNTIME_DETECTION_PATTERN } from '../types/constants';
+import { ForceFunctionStartExecutor } from './ForceFunctionStartExecutor';
+
+/**
+ * Error types when running SFDX: Start Function
+ * This is also used as the telemetry log name.
+ */
+type ForceFunctionStartErrorType = 'force_function_start_docker_plugin_not_installed_or_started';
+
+const forceFunctionStartErrorInfo: {
+  [key in ForceFunctionStartErrorType]: {
+    cliMessage: string;
+    errorNotificationMessage: string;
+  };
+} = {
+  force_function_start_docker_plugin_not_installed_or_started: {
+    cliMessage: 'Cannot connect to the Docker daemon',
+    errorNotificationMessage: nls.localize(
+      'force_function_start_warning_docker_not_installed_or_not_started'
+    )
+  }
+};
+
+export const BINARY_EVENT_ENUM = {
+  CONTAINER: 'container',
+  ERROR: 'error',
+  LOG: 'log',
+  PACK: 'pack'
+};
+
+export class ForceFunctionContainerStartExecutor extends ForceFunctionStartExecutor {
+  // The function binary. It must be used as a single instance. Multiple class to the getFunctionBinary
+  // function will not provide the same instance of the binary.
+  private functionsBinary: Benny | undefined;
+
+  public async setupFunctionListeners(
+    functionDirPath: string,
+    functionDisposable: vscode.Disposable
+  ): Promise<void> {
+    this.functionsBinary = await getFunctionsBinary();
+
+    const writeMsg = (msg: { text: string; timestamp: string }) => {
+      const outputMsg = msg.text;
+
+      if (outputMsg) {
+        channelService.appendLine(outputMsg);
+
+        const matches = String(outputMsg).match(
+          FUNCTION_RUNTIME_DETECTION_PATTERN
+        );
+        if (matches && matches.length > 1) {
```

Comment:
Just curious, does this check need to also check for the presence of matches[1]?  So...
```
if (matches && matches.length > 1 && matches[1]) { ... }
```

Reply:
👍 cleaned it up.